src/main/scala/com/cloudant/clouseau/ClouseauTypeFactory.scala

```diff
@@ -120,49 +120,51 @@ object ClouseauTypeFactory extends TypeFactory {
     result
   }
 
-  private def addFields(doc: Document, field0: Any): Unit = field0 match {
-    case (name: String, value: String, options: List[(String, Any)]) =>
-      val map = options.toMap
-      constructField(name, value, toStore(map), toIndex(map), toTermVector(map)) match {
-        case Some(field) =>
-          map.get("boost") match {
-            case Some(boost: Number) =>
-              field.setBoost(toFloat(boost))
-            case None =>
-              'ok
-          }
-          doc.add(field)
-          if (isFacet(map) && value.nonEmpty) {
-            val fp = FacetIndexingParams.DEFAULT
-            val delim = fp.getFacetDelimChar
-            if (!name.contains(delim) && !value.contains(delim)) {
-              val facets = new SortedSetDocValuesFacetFields(fp)
-              facets.addFields(doc, List(new CategoryPath(name, value)).asJava)
+  private def addFields(doc: Document, field0: Any): Unit = {
+    field0 match {
+      case (name: String, value: String, options: List[_]) =>
```

Comment:
Should not we keep the parameter to the `List[A]` type here?

Reply:
If we would have `List[A]` then newer compiler would be producing warnings like this

```
warning: non-variable type argument (String, Any) in type pattern List[(String, Any)] (the underlying of List[(String, Any)]) is unchecked since it is eliminated by erasure
    case (name: String, value: String, options: List[(String, Any)]) =>
                                                                                  ^
```

New versions of scala do not have ability to match on inner type of containers. It cannot distinguish the following two cases

```scala
   case strings: List[Strings] => ...
   case integers: List[Int] => ...
```

As a result Scala designers decided to introduce the warning. The `List[_]` signals to compiler that user is not trying to match on the type so it is fine use.

Reply:
This also applies to generics. For example I cannot do

```scala
def asListOf[T](input: List[Any]): List[T] = {
    input match {
        case list: List[T] => list.asInstanceOf[List[T]]
        case _ => throw new Exception("")
    } 
}
```

Reply:
Personally I would prefer to keep the type and find a way to disable the warning. Because in this case I know what I am doing. But I couldn't find the way to control the warning. I'll try again.

Reply:
I've found a way

```scala
      case (name: String, value: String, options: List[(String, Any) @unchecked]) => ...
```

Reply:
I'll add the types back and use `@unchecked` trick.

Reply:
Thank you @iilyak for both the explanation and the fix.